tests/unittests/serving/test_experiments_endpoint.py

```diff
@@ -0,0 +1,209 @@
+        """
+        Tests that a 404 response is returned when the experiment
+        doesn't exist in the database
+        """
+        response = client.simulate_get('/experiments/a')
+
+        assert response.status == "404 Not Found"
+        assert response.json['message']
+        assert response.json['message'] == "Experiment 'a' does not exist"
+
+        _add_experiment(name='a', version=1, _id=1)
+        response = client.simulate_get('/experiments/b')
+
+        assert response.status == "404 Not Found"
+        assert response.json['message'] == "Experiment 'b' does not exist"
+
+    def test_experiment_specification(self, client):
+        """Tests that the experiment returned is following the specification"""
+        _add_experiment(name='a', version=1, _id=1)
+        _add_trial(experiment=1, id_override="ae8", status='completed')
+
+        response = client.simulate_get('/experiments/a')
+
+        assert response.status == "200 OK"
+
+        assert response.json['name'] == "a"
+        assert response.json['version'] == 1
+        assert response.json['status'] == "not done"
+        assert response.json['trialsCompleted'] == 1
+        assert response.json['startTime'] == "0001-01-01 00:00:00"  # TODO
+        assert response.json['endTime'] == "0001-01-02 00:00:00"  # TODO
+        assert len(response.json['user'])
+        assert response.json['orionVersion'] == "x.y.z"
+
+        _assert_config(response.json['config'])
+        _assert_best_trial(response.json['bestTrial'])
+
+    def test_default_is_latest_version(self, client):
+        """Tests that the latest experiment is returned when no version parameter exists"""
+        _add_experiment(name='a', version=1, _id=1)
+        _add_experiment(name='a', version=2, _id=2)
+
+        response = client.simulate_get('/experiments/a')
+
+        assert response.status == "200 OK"
+        assert response.json['version'] == 2
+
+    def test_specific_version(self, client):
+        """Tests that the specified version of an experiment is returned"""
+        _add_experiment(name='a', version=1, _id=1)
+        _add_experiment(name='a', version=2, _id=2)
+        _add_experiment(name='a', version=3, _id=3)
+
+        response = client.simulate_get('/experiments/a?version=2')
+
+        assert response.status == "200 OK"
+        assert response.json['version'] == 2
+
+    def test_unknown_parameter(self, client):
+        """
+        Tests that if an unknown parameter is specified in
+        the query string, an error is returned
+        """
+        _add_experiment(name='a', version=1, _id=1)
+
+        response = client.simulate_get('/experiments/a?unknown=true')
+
+        assert response.status == "400 Bad Request"
+        assert response.json == \
+               "Parameter 'unknown' is not supported. Expected parameter 'version'."
+
+
+def _add_experiment(**kwargs):
+    """Adds experiment to the dummy orion instance"""
+    base_experiment.update(kwargs)
```

Comment:
This update may cause nasty side-effects in the future if we are to reuse these methods more globally for tests. We should at least make a deep copy and update the local copy.

Reply:
Sounds good for now! I'll move these when testing the endpoint trials/:experiment so they will be easier to change when reused more globally.